File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/interceptor/LargePictureInterceptor.java
package com.didichuxing.doraemonkit.kit.network.okhttp.interceptor;


import android.text.TextUtils;

import com.didichuxing.doraemonkit.kit.largepicture.LargePictureManager;
import com.didichuxing.doraemonkit.kit.network.okhttp.InterceptorUtil;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * 大图拦截器
 */
public class LargePictureInterceptor implements Interceptor {
    public static final String TAG = "LargePictureInterceptor";


    @NotNull
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Response response = chain.proceed(request);
        String contentType = response.header("Content-Type");

        if (InterceptorUtil.isImg(contentType)) {
            processResponse(response);
        }
        return response;
    }


    private void processResponse(Response response) {
        String field = response.header("Content-Length");
        if (!TextUtils.isEmpty(field)) {
            LargePictureManager.getInstance().process(response.request().url().toString(), Integer.parseInt(field));
        }
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/interceptor/LargePictureInterceptor.kt
package com.didichuxing.doraemonkit.kit.network.okhttp.interceptor

import android.text.TextUtils
import com.didichuxing.doraemonkit.aop.DokitPluginConfig
import com.didichuxing.doraemonkit.config.PerformanceSpInfoConfig.isLargeImgOpen
import com.didichuxing.doraemonkit.kit.largepicture.LargePictureManager
import com.didichuxing.doraemonkit.kit.network.okhttp.InterceptorUtil.isImg
import okhttp3.Interceptor
import okhttp3.Response


/**
 * 大图拦截器
 */
class LargePictureInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val response = chain.proceed(request)
        if (!DokitPluginConfig.SWITCH_BIG_IMG) {
            return response
        }
        val contentType = response.header("Content-Type")

        if (isImg(contentType)) {
            if (isLargeImgOpen()) {
                processResponse(response)
            }
        }
        return response
    }
    private fun processResponse(response: Response) {
        val field = response.header("Content-Length")
        if (!TextUtils.isEmpty(field)) {
            LargePictureManager.getInstance().process(response.request().url().toString(), field!!.toInt())
        }
    }

}